justfile

```diff
@@ -1,29 +1,42 @@
+# Variables
+PYTHON_VERSION := shell("python -c \"import sys;v=f'{sys.version_info.major}.{sys.version_info.minor}';sys.stdout.write(v)\"")
+ENV_DIR := ".env_python" + PYTHON_VERSION
+IN_ENV := ". " + ENV_DIR / "/bin/activate &&"
+PIP_CMD := "python" + PYTHON_VERSION + " -m pip --disable-pip-version-check --no-input --retries 2"
+
 # Recipes
 @default:
   just --list
 
+setup-env *args:
```

Comment:
I'm not sure there's a need to be so prescriptive on the virtualenv location. The `CONTRIBUTING.md` simply shows it as an example. This adds a lot of additional complexity to the project, for the sake of saving a single commad.

Reply:
A bit, though I do appreciate that the abstraction is general enough that if there is a switch to say, ``uv`` over pip in order to provide faster dependency resolution and such, that the switch could be made seamlessly without disturbing or changing any development workflow.